code/espurna/telnet.ino

```diff
@@ -38,16 +47,18 @@ void _telnetWebSocketOnSend(JsonObject& root) {
 #endif
 
 void _telnetDisconnect(unsigned char clientId) {
-    _telnetClients[clientId]->free();
-    delete _telnetClients[clientId];
-    _telnetClients[clientId] = NULL;
+    #if TELNET_SERVER == TELNET_SERVER_WIFICLIENT
+        _telnetClients[clientId].stop();
+    #else
+        _telnetClients[clientId].free();
```

Comment:
As discovered in #1806, this does nothing for the memory:
https://github.com/me-no-dev/ESPAsyncTCP/blob/7e9ed22ed0078097f895dfee6ebcec90046b81b9/src/ESPAsyncTCP.cpp#L216-L222
Just a status method

See the other comment regarding memory management (& possibly unique_ptr )

Reply:
OK, will change it to ``close(true)`` in that case